R/tokens_xptr.R

```diff
@@ -43,6 +60,13 @@ as.tokens_xptr.tokens_xptr <- function(x) {
     rebuild_tokens(result, attrs)
 }
 
+#' @method lengths tokens_xptr
+#' @noRd
+#' @export
+lengths.tokens_xptr <- function(x, use.names = TRUE) {
+    lengths(as.tokens(x))
```

Comment:
You must not use `as.tokens()` because it is very expensive. use `quanteda:::cpp_ntoken()`, which is used in `ntokens.tokens_xptr()`.

Reply:
[Or `ntoken(x))` i](https://github.com/quanteda/quanteda/blob/df6204052012f1db20e2a7933fca0bfb08ac55d9/R/tokens_xptr.R#L80-L83C2)

Reply:
I am not sure if we should have `lengths.tokens_xptr()` because we always recommended users to use quanteda functions, `ntoken()`. It might be even misleading because it make tokens_xptr objects look like list-based objects.

Reply:
I agree that it's not something we should promote, and it's not in the package index. But a list-like object should work with list-like functions.

Reply:
Resolved in [95bd25a](https://github.com/quanteda/quanteda/pull/2255/commits/95bd25a33c16e2119c3cc439963d52b14ce3e869)